src/websocket.js

```diff
@@ -182,11 +168,37 @@ const trades = (payload, cb) => {
         tradeId,
       })
     })
-
-    return w
   })
 
-  return () => cache.forEach(w => w.close())
+  return () => cache.forEach(close => close())
+}
+
+
+const createSocket = (path, cb) => {
+  const onClose = ()=> {
+    if (!closedManually) {
+      w = newSocket(path, cb, onClose)
+    }
+  }
+  
+  let closedManually = false
+  let w = newSocket(path, cb, onClose)  
+  
+  return () => {
+    closedManually = true
+    w.close()
+  }
+}
+
+const newSocket = (path, cb, onClose) => {
+  const w = new WebSocket(`${BASE}/${path}`)
+  const onDisconnect = () => setTimeout(onClose, RECONNECT_DELAY)
```

Comment:
Any reason not to reconnect instantly?

Reply:
Because binance's servers have proven to be awfully unstable and at times down for hours. If we do not apply a delay we can potentially cause a huge overload if their servers are down and we are constantly trying to reconnect. If their servers are down for 1 minute and are immediatly rejecting the connection we can know that with the 1s delay we will be instantiating ~50 websocket instances/attempted connections while if we don't apply a delay we can be looking (and considering it takes 10ms to create a websocket and emit a connection) we will be looking at ~6000 attempted connections in that 1min.

Reply:
True